### DISCUSSION TITLE
Access Vista smart log information

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#32] [Ideas]
Now that the integration is running well, it would be great to have access to the log easily (the one that can be accessed by code+#60) on the vista system. 

What about creating a smart log? 

Home assistant could pull the last 10 or so (even better if we can get to 100) log items once a refresh key is pressed and display then as a log in plain English as the codes are defined in the manual. Even better, if the user number could also be changed to a person’s name (adding a way to defined each users).

**Comment by AlexandreUSA** [comment]
I am using the Home Assistant logbook lovelace card and it seems to do what I need. Would be nice to have a way to rename "user" (e.g. user 3) with a real name.

**Reply by Dilbert66** [reply]
Can you post an example of the setup you use for the logbook card? Could be helpful for other users.

**Comment by Dilbert66** [comment]
I will look into it and see what I can find. I don't want to put too much logic in the ESP, especially if some of the heavy lifting can be done in HA itself.  What I try to do is have the esp code send enough info for HA to do it's own decisions.  For example, you can have some logic in HA that does it's own lookup for user codes.

**Reply by AlexandreUSA** [reply]
Sounds good. I will play with HA and see what I can do. I am getting the message from LRR in HA so I should be able to get something out of it. I am still new in HA so I am still learning. So far, it is working very well.

**Reply by AlexandreUSA** [reply]
Right now, I just added the standard "logbook" card in HA and I added the "sensor.vistaalarm_lrr_msg" sensor. I had also added the RFX messages but they are not very useful so I removed them. I have not tried yet to see what happens when there is an alarm. I will put my system on test with the monitoring station later this week or this weekend so I can run additional tests. I added few smoke/CO detectors that I need to test.

This is the card. Some of the old entries are duplicate due to reboots (now fixed)
![log](https://user-images.githubusercontent.com/56739736/121384200-0b3de000-c916-11eb-8ebe-1df863422325.JPG)

My next step is to setup a push notification with zone triggered and a link to my cameras when there is an alarm/fire. The monitoring company will call the house/cell (but when I travel back to Europe or if I am out of town, better can a notification too).

**Comment by AlexandreUSA** [comment]
I am getting closer to where I want to be with my log using the Lovelace Home Feed Card. A little bit of clean up but here is a start:
![image](https://user-images.githubusercontent.com/56739736/122599848-fa3f4e00-d03c-11eb-891a-d204121b27b4.png)

**Reply by AlexandreUSA** [reply]
By the way, by remapping the state to new names in the custom card, I am wondering what the O/C means? Why O/C by user user # statement

```
state_map:
      '441: Armed STAY user 5 ': <b><font color="orange">Armed STAY</font></b> by <b>Casey</b>
      '401: O/C by user user 5 ': <b><font color="orange">Armed AWAY</font></b> by <b>Casey</b>
      '441: Armed STAY user 5 Restored': <b><font color="green">Disarmed STAY</font></b> by <b>Casey</b>
      '401: O/C by user user 5 Restored': <b><font color="green">Disarmed AWAY</font></b> by <b>Casey</b>
```
in reference probably to 
`case 401: return F("UO/C by user");`

Also, just a tiny thing but I need to add a space after "user # ' for arming name but not after "user # restored". Is there an extra space somewhere in the vistaalarm code?

**Reply by AlexandreUSA** [reply]
If another user needs the reason why there is an extra space while doing comparison on the state value for the LRR messages, here it is: 
Line 630 in vistaalarm.h: 
`sprintf(msg,"%d: %s %s %d %s",c, &lrrString[1],uf.c_str(),z,qual.c_str());`
It makes sense as when the qual.str() is blank, we will get that extra space between uf and qual.

**Reply by Dilbert66** [reply]
I can fix that by adding the space to the restore/cleared text instead so that when those terms are not used, there will not be a hanging space at the end.

**Reply by AlexandreUSA** [reply]
By the way, there may be also an issue with the display of zone numbers with two digits:
For example, if I bypass zone 12, I get:

`570: Zone/Sensor bypass zone 2 Active`

I did not try with users as I do not have any user # with double digits

**Reply by Dilbert66** [reply]
You are correct. There is a bug. I will correct it and re-upload shortly.

**Reply by Dilbert66** [reply]
Updated master with corrected version.  Removed extra space, corrected missing zone digit and change the 401 message to Armed Away

**Reply by AlexandreUSA** [reply]
Thanks for getting this done so quickly.  Working well now. I will continue playing with the integration with HA. You are doing an amazing job.

**Comment by Dilbert66** [comment]
Using standard alarm CID reporting codes: Example list: https://zionssecurity.com/contact-id-codes/.
O/C = open/close.  I took the list word for word from a list I had. Not this one but another.

**Reply by AlexandreUSA** [reply]
Got it. Thanks for the reply. I am using the Ademco/Vista list in the manual. That explains some of the differences such as the 401 being the Armed/Disarmed for AWAY/MAXIMUM while on your list, it is O/C.

![ContactID](https://user-images.githubusercontent.com/56739736/122602953-a5ea9d00-d041-11eb-82b3-29be567545e9.jpg)

**Comment by Dilbert66** [comment]
I never tested all reporting codes. Yes, it still needs some tweaks to adjust to the Honeywell use of them.

**Reply by AlexandreUSA** [reply]
I am renaming them to be more user friendly so not an issue on my side. Looking at the lists, it seems that 401 is the main difference anyway. The rest seems very standard.
